### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -44,7 +44,11 @@ class TestnetNode {
     return Promise.all(promises)
   }
   stop () {
-    return Promise.resolve()
+    const promises = []
+    if (this.config.btp) {
+      promises.push(this.pluginFactory.stop())
```
Comment (sentientwaffle) [inline src/index.js]: If there's only one promise, why use an array and `Promise.all()`?

File: src/quoter.js [hunk 2]
```diff
@@ -83,28 +83,34 @@ Quoter.prototype = {
   },
 
   answerLiquidity (req) {
-    const curve = this.findCurve(req.destinationAccount)
-    return Promise.resolve({
-      liquidityCurve: curve.buf,
-      appliesToPrefix: curve.prefix,
-      sourceHoldDuration: 15000,
-      expiresAt: new Date(Date.now() + 3600 * 1000)
+    return Promise.resolve().then(() => {
```
Comment (sentientwaffle) [inline src/quoter.js:86]: What is the reason for this change?

File: src/transfer-handler.js [hunk 3]
```diff
@@ -11,23 +14,37 @@ class TransferHandler {
...
+            noteToSelf: {},
+            executionCondition: transfer.executionCondition,
+            expiresAt: new Date(new Date(transfer.expiresAt).getTime() + MIN_MESSAGE_WINDOW).toISOString(),
+            custom: {}
+          })).catch(err => {
+            console.error('forwarding failed', err.message)
+          })
+        } else {
+          this.main.getPlugin(prefix).rejectIncomingTransfer(transfer, {
+            code: 'L53',
```
Comment (sentientwaffle) [inline src/transfer-handler.js:44]: Is there an appropriate error code from https://github.com/interledger/rfcs/blob/master/0003-interledger-protocol/0003-interledger-protocol.md#ilp-error-codes that could be used?

**Review by sentientwaffle — CHANGES_REQUESTED** [review]

File: src/quoter.js (hunk 2 above)
Comment (michielbdejong) [reply src/quoter.js:86]: Because https://github.com/interledgerjs/amundsen/blob/master/src/request-handler.js#L34 requires this function to return a promise.

File: src/transfer-handler.js (hunk 3 above)
Comment (michielbdejong) [reply src/transfer-handler.js:44]: Not really I think, which one were you thinking of?

File: src/index.js (hunk 1 above)
Comment (michielbdejong) [reply src/index.js]: I agree it's currently superfluous, but I did it this way for symmetry with the start() function. Also in case we want to add a this.removePlugin function in the future.

File: src/quoter.js (hunk 2 above)
Comment (sentientwaffle) [reply src/quoter.js:86]: That's not what I mean.

```
return Promise.resolve().then(() => {
    return Promise.resolve(foo)
})
```

is redundant. You can just ```return Promise.resolve(foo)``` for the same effect.

File: src/transfer-handler.js (hunk 3 above)
Comment (sentientwaffle) [reply src/transfer-handler.js:44]: I didn't have one in mind, but the `"L"` prefix isn't used anywhere afaik. It should either be `"F"`, `"T"`, or `"R"` depending on what type of error it is. And probably `99` to indicate it is an application error.

File: src/quoter.js (hunk 2 above)
Comment (michielbdejong) [reply src/quoter.js:86]: corrected.

File: src/transfer-handler.js (hunk 3 above)
Comment (michielbdejong) [reply src/transfer-handler.js:44]: This code no longer exists.

**Comment by michielbdejong** [comment]
Comments addressed and changes merged in #22
